### PR TITLE
Add encoder decoder utilization metrics

### PR DESCRIPTION
**Pull request by fatsheep9146**
This repo is a great job! 

Since our company use nvidia gpu with ffmpeg library to encode and decode videos, so we need to add functions to get metrics about encoder utilization and decoder utilization.

### CODE REVIEW

File: bindings.go [hunk 1]
```diff
@@ -454,6 +478,26 @@ func (d Device) AverageGPUUtilization(since time.Duration) (uint, error) {
 	return uint(n), errorString(r)
 }
```
Comment (mindprince) [inline bindings.go]: Can you add doc comments? What are the two values getting returned.

File: bindings.go [hunk 2]
```diff
@@ -454,6 +478,26 @@ func (d Device) AverageGPUUtilization(since time.Duration) (uint, error) {
...
+func (d Device) EncoderUtilization() (uint, uint, error) {
+	if C.nvmlHandle == nil {
+		return 0, 0, errLibraryNotLoaded
+  }
+  var n C.uint
+  var sp C.uint
+  r := C.nvmlDeviceGetEncoderUtilization(d.dev, &n, &sp)
+  return uint(n), uint(sp), errorString(r) 
+}
+
```
Comment (mindprince) [inline bindings.go]: Can you add doc comments? What are the two values getting returned.

Comment (fatsheep9146) [reply bindings.go]: OK, I will add doc comments :)

**Comment by fatsheep9146** [comment]
Done @mindprince

**Comment by cofyc** [comment]
@mindprince Is it possible to merge this PR? We hope [cadvisor](https://github.com/google/cadvisor) can report GPU utilization metrics.

**Comment by mindprince** [comment]
@cofyc Merged this PR but it's hard to get new GPU specific metrics into cadvisor. We are working towards an extension mechanism that people can use instead.

**Comment by fatsheep9146** [comment]
What is this extension mechanism about? Can you list some related issues about this extension mechanism, thank you!

**Comment by mindprince** [comment]
@fatsheep9146 The extension mechanism is to expose some metadata from kubelet, so that agents outside the kubelet can monitor devices etc. and then associate these device metrics with containers and pods. I don't think there's any issues for this yet. Just a lot of discussion in the [Resource Management WG](https://github.com/kubernetes/community/tree/master/wg-resource-management)

**Comment by fatsheep9146** [comment]
@mindprince Thanks!

**Pull request merged**